src/v/kafka/requests/metadata_request.cc

```diff
@@ -360,11 +361,19 @@ get_topic_metadata(request_context& ctx, metadata_request& request) {
     std::vector<ss::future<metadata_response::topic>> new_topics;
 
     for (auto& topic : *request.topics) {
+        auto mts = model::materialized_or_source_topic(topic);
         if (auto md = ctx.metadata_cache().get_topic_metadata(
-              model::topic_namespace_view(cluster::kafka_namespace, topic));
+              model::topic_namespace_view(cluster::kafka_namespace, mts));
             md) {
-            res.push_back(metadata_response::topic::make_from_topic_metadata(
-              std::move(*md)));
+            auto mdr_topic = metadata_response::topic::make_from_topic_metadata(
+              std::move(*md));
+            if (mts != topic) {
```

Comment:
Seems a bit haskish to me, open to suggestions here

Reply:
i wonder if materialized topics should instead use a `<parent>.__rp__.<child_stream>` pattern which will make everything readable and compliant. Thoughts? or maybe instead of dollar signs you use `Z` so it becomes `<parent>Z<child>Z`

or maybe both! haha `<parent>.Z__rp__Z.<child>`

Reply:
i vote for the first one. or maybe simply `<parent>.__redpanda__.<child>` which probably make more sense tbh.

Reply:
lol either way the point would be to agree on a schema that doesn't use invalid kafka topic characters, right?

Reply:
is this resolved? im not sure how the naming would affect this logic where the client expects the original topic name.

Reply:
i think this was just discussion, my comment about 'this being a bit hackish' wasn't about the materialized topic scheme, it was about the if statement randomly placed in there